sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -261,20 +224,43 @@ def incr_mean_variance_axis0(X, last_mean, last_var, last_n):
     """
     if X.dtype not in [np.float32, np.float64]:
         X = X.astype(np.float64)
-    return _incr_mean_variance_axis0(X.data, X.shape[0], X.shape[1], X.indices,
-                                     X.indptr, X.format, last_mean, last_var,
-                                     last_n)
+    X_dtype = X.dtype
+    if weights is None:
+       weights = np.ones(X.shape[0], dtype=X_dtype)
+    elif weights.dtype not in [np.float32, np.float64]:
+        weights = weights.astype(np.float64, copy=False)
+    if last_n.dtype not in [np.float32, np.float64]:
+        last_n = last_n.astype(np.float64, copy=False)
+
+    ind_rows, ind_cols, X_data = sp.find(X)
+
+    return _incr_mean_variance_axis0(X_data,
+                                     np.sum(weights),
+                                     X.shape[1],
+                                     ind_rows,
+                                     ind_cols,
+                                     X.indptr,
+                                     X.format,
+                                     last_mean.astype(X_dtype, copy=False),
+                                     last_var.astype(X_dtype, copy=False),
+                                     last_n.astype(X_dtype, copy=False),
+                                     weights.astype(X_dtype, copy=False))
 
 
 def _incr_mean_variance_axis0(np.ndarray[floating, ndim=1] X_data,
-                              unsigned long long n_samples,
+                              floating n_samples,
                               unsigned long long n_features,
-                              np.ndarray[integral, ndim=1] X_indices,
+                              # find() returns int32 so we can set them to long
+                              np.ndarray[int, ndim=1] X_row_ind,
+                              np.ndarray[int, ndim=1] X_indices,
+                              # X_indptr might be either in32 or int64
                               np.ndarray[integral, ndim=1] X_indptr,
```

Comment:
I think that X_indptr is not used anymore. Implicitly you make a copy of X to cast it to COO with sp.find.

do you confirm?

Reply:
yes. you are right 
https://github.com/scipy/scipy/blob/01d8bfb6f239df4ce70c799b9b485b53733c9911/scipy/sparse/extract.py#L36